containers/daap-python-base/src/var/task/data_product_metadata.py

```diff
@@ -495,3 +473,83 @@ def _detect_update_type(self) -> UpdateType:
     def add_auto_generated_metadata(self):
         """adds key value pairs to metadata that are not given by a user."""
         raise NotImplementedError
+
+
+class VersionCreator:
+    """
+    Service to create new versions of a data product when metadata or schema are updated.
+    """
+
+    def __init__(self, data_product_name, logger: DataPlatformLogger):
+        self.data_product_config = DataProductConfig(name=data_product_name)
+        self.logger = logger
+
+    def update_metadata(self, input_data) -> str:
+        """
+        Create a new version with updated metadata.
+        """
+        metadata = DataProductMetadata(
+            data_product_name=self.data_product_config.name,
+            logger=self.logger,
+            input_data=input_data,
+        ).load()
+        if not metadata.valid:
+            raise InvalidUpdate()
+
+        state = metadata.detect_update_type()
+        self.logger.info(f"Update type {state}")
+
+        if state != UpdateType.MinorUpdate:
+            raise InvalidUpdate(state)
+
+        latest_version = metadata.version
+        new_version = metadata.generate_next_version_string()
+        self._copy_from_previous_version(
```

Comment:
This is the new part. This first copies the whole metadata directory from the previous version so we can then overwrite the new metadata.